### ISSUE TITLE
NullReferenceException when calling ShouldHaveChildValidator

### ISSUE THREAD

**Issue by Broccooli** [JeremySkinner/FluentValidation#685]
There's an NRE thrown when calling `ShouldHaveChildValidator()` when the validator on the object is of the form `RuleFor(x => x).SetValidator(someValidator)`.

This is because the `expression.GetMember()` extenstion method returns null when the expression cannot be cast to a `MemberExpression`

Here's the code in question:
```csharp
public static MemberInfo GetMember(this LambdaExpression expression) 
{
    var memberExp = RemoveUnary(expression.Body) as MemberExpression;
    if (memberExp == null) {
        return null;
    ...
}
```

and the calling code that throws the exception:

```csharp
public static void ShouldHaveChildValidator<T, TProperty>(this IValidator<T> validator, Expression<Func<T, TProperty>> expression, Type childValidatorType)
 {
...
    var expressionMemberName = expression.GetMember().Name
...
}
```

I am not sure if this is by design, or an over-sight in implementation. I'm aware that the method itself is named in such a way that it would imply only Members can be gotten from it, but I'd like to see some support for testing validators that are attached via `RuleFor(x => x).SetValidator(...)` or at least some robustness in failure indicating why the test fails.

Perhaps a method `ShouldHaveValidator<T>(...)` could be implemented to cover this use-case.

**Comment by JeremySkinner** [comment]
Apologies for not replying sooner. The ShouldHaveChildValidator extension is designed for use with rules defined against a property. FluentValidation did not used to support rules at the model-level with (x=>x), and the test helper was never updated to be in line with the rest of the project when support for this was added. 

I'll look at expanding this for a possible future release.

Ps, Personally I don't consider using the ShouldHaveChildValidator method good practice - when testing validators it's better to test the behaviour rather than the implementation as it leads to brittle tests.

**Comment by JeremySkinner** [comment]
I've released 7.5 to nuget today that includes this fix.
